coc/static/update_static.py

```diff
@@ -6,229 +6,1482 @@
+        full_super_troop_data = self.open_file("super_licences.json")
+        full_super_troop_data = {v.get("Replacement"): v for k, v in full_super_troop_data.items()}
+
+        name_to_id = {}
+        new_troop_data = []
+        for _id, (troop_name, troop_data) in enumerate(self.full_troop_data.items(), 4000000):
+            if troop_data.get("DisableProduction", False):
+                continue
+            village_type = troop_data.get("VillageType", 0)
+            production_building = self.full_building_data.get(troop_data.get("ProductionBuilding")).get("TID")
+
+            name_to_id[(troop_name, village_type)] = _id
+            hold_data = {
+                "_id": _id,
+                "name": self._translate(tid=troop_data.get("TID")),
+                "info": self._translate(tid=troop_data.get("InfoTID")),
+                "TID": {
+                    "name": troop_data.get("TID"),
+                    "info": troop_data.get("InfoTID"),
+                },
+                "production_building": self._translate(tid=production_building),
+                "production_building_level": troop_data.get("BarrackLevel"),
+                "upgrade_resource": self._parse_resource(resource=troop_data.get("UpgradeResource")),
+
+                "is_flying": troop_data.get("IsFlying"),
+                "is_air_targeting": troop_data.get("AirTargets"),
+                "is_ground_targeting": troop_data.get("GroundTargets"),
+
+                "movement_speed": troop_data.get("Speed", 0),
+
+                "attack_speed": troop_data.get("AttackSpeed", 0),
+                "attack_range": troop_data.get("AttackRange", 0),
+                "housing_space": troop_data.get("HousingSpace"),
+                "village": "home" if not village_type else "builderBase",
+            }
+            is_super_troop = troop_data.get("EnabledBySuperLicence", False)
+            is_seasonal_troop = troop_data.get("EnabledByCalendar", False)
+            super_troop_data = None
+            if is_super_troop:
+                super_troop_data = full_super_troop_data.get(troop_name)
+                hold_data["super_troop"] = {"original_id": name_to_id[(super_troop_data["Original"], 0)],
+                                            "original_min_level": super_troop_data["MinOriginalLevel"]}
+            if is_seasonal_troop:
+                hold_data["is_seasonal"] = True
+            hold_data["levels"] = []
+
+            max_townhall_converter = self.lab_to_townhall
+            if troop_data.get("ProductionBuilding") == "Barrack2":
+                max_townhall_converter = self.bb_lab_to_townhall
+
+            for level, level_data in troop_data.items():
+                if not isinstance(level_data, dict):
+                    continue
+                #convert times to seconds, all times for all things will be in seconds
+                upgrade_time_seconds = level_data.get("UpgradeTimeH", 0) * 60 * 60
```

Comment:
same here about the upgradetime, let's make it as safe as possible and include seconds, minutes and days